net-utils/src/lib.rs

```diff
@@ -384,29 +384,67 @@ pub fn is_host_port(string: String) -> Result<(), String> {
     parse_host_port(&string).map(|_| ())
 }
 
+#[derive(Clone, Debug)]
```

Comment:
kinda regretting not deriving `Copy` after typing `.clone()` more times than anticipated...